src/main/java/org/opensearch/timeseries/util/ParseUtils.java

```diff
@@ -515,34 +530,50 @@ public static void getDetector(
         }
     }
 
-    public static void onGetAdResponse(
+    /**
+     * This method processes the GetResponse and applies the consumer function if the user has permissions
+     * or if the filterByBackendRole is disabled. It uses a ConfigFactory to parse the correct type of Config.
```

Comment:
What is the expected behavior if the filterByBackendRole is disabled? Maybe add the behaviors here?

Reply:
We will not check backend role related user permission if  if the filterByBackendRole is disabled. Yeah, will add.